File: src/main/java/com/golem/lab4/data/Dot.java
package com.golem.lab4.data;

import jakarta.persistence.*;

@Entity
@Table(name = "dot")
public class Dot {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "dot_seq")
    @SequenceGenerator(name = "dot_seq", sequenceName = "dot_seq")
    private Long id;
    @Column(name = "x")
    private double x;
    @Column(name = "y")
    private double y;
    @Column(name = "r")
    private double r;
    @Column(name = "hit")
    private boolean hit;

    public Dot() {}
    public void setX(double x) {
        this.x = x;
    }

    public void setY(double y) {
        this.y = y;
    }

    public void setR(double r) {
        this.r = r;
    }

    public void setHit(boolean hit) {
        this.hit = hit;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public double getR() {
        return r;
    }

    public boolean getHit () {
        return hit;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }
}

File: src/main/webapp/my-app/src/components/Dot.js
class Dot {
    x
    y
    r
    hit
    constructor(x, y, r) {
        this.x = x;
        this.y = y;
        this.r = r;
        this.hit = this.checkHit(x, y, r);
    }

    checkHit(x, y, r) {
        return x >= 0
            ? y >= 0
                ? -x + r > y
                : false
            : y >= 0
                ? x*x + y*y <= r*r
                : x >= -r/2 && y >= -r;
    }
}

export default Dot